CONTRIBUTING.md

```diff
@@ -0,0 +1,110 @@
+# Contributing
+
+## Development
+
+### System Requirements
+
+Python 3.8 and above are required.
+
+### Target version(s)
+
+Python 3.8 and above are supported by the SDK.
+
+### Installation and Dependencies
+
+A [`Makefile`](./Makefile) has been included in the project which should make it straightforward to start the project locally. We utilize virtual environments (see [`virtualenv`](https://docs.python.org/3/tutorial/venv.html)) in order to provide isolated development environments for the project. This reduces the risk of invalid or corrupt global packages. It also integrates nicely with Make, which will detect changes in the `requirements-dev.txt` file and update the virtual environment if any occur.
+
+Run `make init` to initialize the project's virtual environment and install all dev dependencies.
+
+### Testing
+
+Run tests with `make test`.
+
+We use `pytest` for our unit testing, making use of `parametrized` to inject cases at scale. 
+
+### Integration tests
+
+These are planned once the SDK has been stabilized and a Flagd provider implemented. At that point, we will utilize the [gherkin integration tests](https://github.com/open-feature/test-harness/blob/main/features/evaluation.feature) to validate against a live, seeded Flagd instance.
+
+### Packaging
+
+We publish to the PyPI repository, where you can find this package at [openfeature-sdk](https://pypi.org/project/openfeature-sdk/).
+
+## Pull Request
+
+All contributions to the OpenFeature project are welcome via GitHub pull requests.
+
+To create a new PR, you will need to first fork the GitHub repository and clone upstream.
+
+```bash
+git clone https://github.com/open-feature/python-sdk.git openfeature-python-sdk
+```
+
+Navigate to the repository folder
+
+```bash
+cd openfeature-python-sdk
+```
+
+Add your fork as an origin
+
+```bash
+git remote add fork https://github.com/YOUR_GITHUB_USERNAME/python-sdk.git
+```
+
+Makes sure your development environment is all setup by building and testing
+
+```bash
+make
+```
+
+To start working on a new feature or bugfix, create a new branch and start working on it.
+
+```bash
+git checkout -b feat/NAME_OF_FEATURE
```

Comment:
Should we precede this step with steps to check out main -- I use git fetch --all --prune then git checkout upstream/main but perhaps there is a better way.

Perhaps everyone contributing is sophisticated to know this, but I've been amazed over the years at how many people don't realize they need to do this or their branch will include the changes from the branch they worked on last.

Reply:
Since the instructions are written in the context of having just cloned the repository to start the work, so I'm not sure that adding additional steps would make sense since they are unnecessary in this context

Reply:
I tend to agree to with @tcarrio but thanks for the comment @StackScribe .